WordPress/Classes/Services/PostHelper.m

```diff
@@ -92,6 +92,7 @@ + (void)updatePost:(AbstractPost *)post withRemotePost:(RemotePost *)remotePost
             publicizeMessage = [publicizeMessageDictionary stringForKey:@"value"];
             publicizeMessageID = [publicizeMessageDictionary stringForKey:@"id"];
         }
+        postPost.foreignID = remotePost.foreignID;
```

Comment:
This code runs only for posts. Pages need the same mechanism.

Reply:
Good catch, fixed